docs/examples/_data_server.py

```diff
@@ -0,0 +1,135 @@
+import asyncio
+
+from grpclib.utils import graceful_exit
+from grpclib.server import Server, Stream
+from google.protobuf.struct_pb2 import Struct
+
+from viam.proto.app.data import (
+    AddBoundingBoxToImageByIDResponse,
+    AddBoundingBoxToImageByIDRequest,
+    AddTagsToBinaryDataByFilterRequest,
+    AddTagsToBinaryDataByFilterResponse,
+    AddTagsToBinaryDataByIDsRequest,
+    AddTagsToBinaryDataByIDsResponse,
+    BinaryDataByFilterRequest,
+    BinaryDataByFilterResponse,
+    BinaryDataByIDsRequest,
+    BinaryDataByIDsResponse,
+    BoundingBoxLabelsByFilterRequest,
+    BoundingBoxLabelsByFilterResponse,
+    DataServiceBase,
+    DeleteBinaryDataByFilterRequest,
+    DeleteBinaryDataByFilterResponse,
+    DeleteBinaryDataByIDsRequest,
+    DeleteBinaryDataByIDsResponse,
+    DeleteTabularDataByFilterRequest,
+    DeleteTabularDataByFilterResponse,
+    RemoveBoundingBoxFromImageByIDResponse,
+    RemoveBoundingBoxFromImageByIDRequest,
+    RemoveTagsFromBinaryDataByFilterRequest,
+    RemoveTagsFromBinaryDataByFilterResponse,
+    RemoveTagsFromBinaryDataByIDsRequest,
+    RemoveTagsFromBinaryDataByIDsResponse,
+    TabularData,
+    TabularDataByFilterRequest,
+    TabularDataByFilterResponse,
+    TagsByFilterRequest,
+    TagsByFilterResponse,
+)
+from viam.proto.app.datasync import (
+    DataCaptureUploadRequest,
+    DataCaptureUploadResponse,
+    DataSyncServiceBase,
+    FileUploadRequest,
+    FileUploadResponse,
+)
+
+
+class DataServer(DataServiceBase, DataSyncServiceBase):
```

Comment:
I'm not entirely certain I understand what this class is doing. Can you clarify?

Reply:
Yes! This is another mock data service (similar to the one that had to be created in the testing ticket). This one is used for the example documentation. This server runs on a local port and pretends to be app and feeds our example data_client--the one included in the example notebook and so displayed on our docs page--fake data just to display. This is used because we can't actually authenticate and connect to app because we don't want to include a real location secret or robot uri in the docs pages or in the python repo.

Put differently, the example page writes a function that shows how to instantiate an AppClient. We don't actually call this function though, because we know it would fail since we aren't providing real location secrets or URIs. Instead, I depend on a fake mock AppClient that directly connects its DataClient to this server^ to give the reader the illusion of a data client. Any code that would 'reveal' this illusion is marked to be hidden and so won't show up on the html page, but it's still in the repo (and still runs and shouldn't cause any error at all).

That is all!